cmd/kubeadm/app/cmd/init.go

```diff
@@ -154,6 +154,32 @@ func NewCmdInit(out io.Writer) *cobra.Command {
 			// TODO: the code in runInit should be progressively converted in phases; each phase will be exposed
 			// via the subcommands automatically created by initRunner.BindToCommand
 			err = runInit(&data, out)
+
+			// printing the join command should happen after all the phases in init have finished
+			showJoinCommand := func(i *initData, out io.Writer) error {
```

Comment:
let's not have it as a anonymous function but rather define it in the file level:
```
func showJoinCommand(....) {
```